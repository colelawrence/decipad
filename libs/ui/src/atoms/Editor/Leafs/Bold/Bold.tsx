import { css } from '@emotion/react';
import { PlatePluginComponent } from '@udecode/plate';

const styles = css({
  fontWeight: 'bold',
});

export const BoldLeaf: PlatePluginComponent = ({ attributes, children }) => {
  return (
    <span css={styles} {...attributes}>
      {children}
    </span>
  );
};
