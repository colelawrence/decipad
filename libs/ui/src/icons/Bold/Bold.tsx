import { FC } from 'react';
import { cssVar } from '../../primitives';

export const Bold = (): ReturnType<FC> => {
  return (
    <svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
      <title>Bold</title>
      <path
        d="M4 2.66669H9.33333C10.0406 2.66669 10.7189 2.94764 11.219 3.44774C11.719 3.94783 12 4.62611 12 5.33335C12 6.0406 11.719 6.71888 11.219 7.21897C10.7189 7.71907 10.0406 8.00002 9.33333 8.00002H4V2.66669Z"
        stroke={cssVar('weakTextColor')}
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M4 8H10C10.7072 8 11.3855 8.28095 11.8856 8.78105C12.3857 9.28115 12.6667 9.95942 12.6667 10.6667C12.6667 11.3739 12.3857 12.0522 11.8856 12.5523C11.3855 13.0524 10.7072 13.3333 10 13.3333H4V8Z"
        stroke={cssVar('weakTextColor')}
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
};
