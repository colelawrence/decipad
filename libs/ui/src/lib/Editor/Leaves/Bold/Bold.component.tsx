import { Text } from '@chakra-ui/react';

import { RenderLeafProps } from 'slate-react';

export const Bold = ({
  attributes,
  children,
}: RenderLeafProps): JSX.Element => {
  return (
    <Text as="span" fontWeight="bold" {...attributes}>
      {children}
    </Text>
  );
};
