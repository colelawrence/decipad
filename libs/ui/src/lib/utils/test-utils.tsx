import { ChakraProvider } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';
import React from 'react';

import { Type } from '@decipad/language';
import { theme, ResultsContextProvider } from '../..';

const results = {
  test: {
    type: Type.build({
      type: 'number',
      unit: {
        unit: 'apples',
      },
    } as any),
    value: [20],
    errors: [],
  },
  noResult: null,
};

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider resetCSS theme={theme}>
      <ResultsContextProvider value={results}>
        {children}
      </ResultsContextProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
