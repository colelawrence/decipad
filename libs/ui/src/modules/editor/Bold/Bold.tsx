import { css } from '@emotion/react';
import { ReactNode } from 'react';
import { p16Bold } from '../../../primitives';

const styles = css(p16Bold, {
  fontStyle: 'inherit',
  fontSize: 'inherit',
  lineHeight: 'inherit',
});

interface BoldProps {
  readonly children: ReactNode;
}
export const Bold = ({ children }: BoldProps): ReturnType<React.FC> => {
  return <span css={styles}>{children}</span>;
};
