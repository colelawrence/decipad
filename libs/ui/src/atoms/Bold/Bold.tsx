import { css } from '@emotion/react';
import { ReactNode } from 'react';
import { p16Bold } from '../../primitives';
import { SlateLeafProps } from '../../utils';

const styles = css(p16Bold);

interface BoldProps extends SlateLeafProps {
  readonly children: ReactNode;
}
export const Bold = ({ children }: BoldProps): ReturnType<React.FC> => {
  return <span css={styles}>{children}</span>;
};
