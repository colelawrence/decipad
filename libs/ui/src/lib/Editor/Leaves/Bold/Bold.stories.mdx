import { Meta, Story, Canvas, Props } from '@storybook/addon-docs/blocks';
import { Bold } from './Bold.component';

<Meta
  title="Legacy/Editor/Leaves/Bold"
  parameters={{
    docs: {
      source: {
        code: `<Bold
  attributes={{ 'data-slate-leaf': true }}
  leaf={{ text: 'Bold text' }}
  text={{ text: 'Bold text' }}
>
  Bold text
</Bold>`,
      },
    },
  }}
  argTypes={{
    children: {
      defaultValue: 'Hello World',
      description: 'The text content of the component',
      type: 'string',
    },
  }}
  component={Bold}
/>

# Bold

The bold leaf makes any text well...bold.

export const Template = (args) => <Bold {...args} />;

<Canvas>
  <Story name="Bold">{Template.bind({})}</Story>
</Canvas>

<Props story="Bold" />
