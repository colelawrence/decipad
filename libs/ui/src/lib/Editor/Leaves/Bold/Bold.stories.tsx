import React from 'react';
import { Bold } from './Bold.component';

export default {
  title: 'Editor/Leaves/Bold',
};

export const B = () => (
  <Bold
    text={{ text: 'hello world' }}
    leaf={{ text: 'hello world' }}
    attributes={{
      'data-slate-leaf': true,
    }}
  >
    The quick brown fox jumps over the lazy dog
  </Bold>
);

B.storyName = 'Bold';
