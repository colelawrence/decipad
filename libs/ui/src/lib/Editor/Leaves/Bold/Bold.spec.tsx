import React from 'react';
import { render } from 'test-utils';
import { Bold } from './Bold.component';

describe('Bold Leaf', () => {
  it('renders', () => {
    const { container } = render(
      <Bold
        attributes={{ 'data-slate-leaf': true }}
        leaf={{ text: 'Bold text' }}
        text={{ text: 'Bold text' }}
      >
        Bold text
      </Bold>
    );

    expect(container).toBeDefined();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <Bold
        attributes={{ 'data-slate-leaf': true }}
        leaf={{ text: 'Bold text' }}
        text={{ text: 'Bold text' }}
      >
        Bold text
      </Bold>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders the string', () => {
    const { getByText } = render(
      <Bold
        attributes={{ 'data-slate-leaf': true }}
        leaf={{ text: 'Bold text' }}
        text={{ text: 'Bold text' }}
      >
        Bold text
      </Bold>
    );

    getByText('Bold text');
  });
});
