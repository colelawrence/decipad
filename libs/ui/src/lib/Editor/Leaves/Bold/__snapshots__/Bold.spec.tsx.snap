// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bold Leaf matches snapshot 1`] = `
.emotion-0 {
  font-weight: var(--chakra-fontWeights-bold);
}

<div>
  <span
    class="chakra-text emotion-0"
    data-slate-leaf="true"
  >
    Bold text
  </span>
</div>
`;
